src/html-transform.ts

```diff
@@ -102,52 +123,89 @@ export function htmlTransform(
     }
   }
 
-  if (shouldTransformEsModuleToAmd) {
-    // We need RequireJS to load the AMD modules we are declaring. Inject the
-    // dependency as late as possible (just before the first module is
-    // declared) because there may be some UMD dependencies that we want to
-    // continue to load in global mode instead of AMD mode (which is detected
-    // by the presence of the `require` global).
-    if (firstModuleScript !== undefined && options.isEntryPoint) {
-      const fragment = parse5.parseFragment(
-          `<script src="${options.requireJsUrl}"></script>\n`);
-      const requireJsScript = fragment.childNodes![0];
+  if (shouldTransformEsModuleToAmd && finalModuleScript !== undefined) {
+    // We've defined a bunch of modules and chained them together. Now we need
+    // to initiate loading the chain by requiring the final one.
+    const finalModuleName = generateModuleName(moduleScriptIdx - 1);
+    const fragment = parse5.parseFragment(
+        `<script>require(['${finalModuleName}']);</script>`);
+    dom5.insertAfter(
+        finalModuleScript.parentNode!, finalModuleScript, fragment);
+  }
+
+  if (options.injectAmdLoader) {
+    const fragment = parse5.parseFragment('<script></script>\n');
+    dom5.setTextContent(fragment.childNodes![0], getMinifiedRequireJs());
+    const requireJsScript = fragment.childNodes![0];
+
+    // Inject as late as possible (just before the first module is declared, if
+    // there is one) because there may be some UMD dependencies that we want to
+    // continue to load in global mode instead of AMD mode (which is detected by
+    // the presence of the `require` global).
+    if (firstModuleScript !== undefined) {
       dom5.insertBefore(
           firstModuleScript.parentNode!, firstModuleScript, fragment);
+    } else {
+      // TODO(aomarks) If there were no modules, where do we put the AMD loader?
```

Comment:
we don't need to?

Reply:
I was thinking that you could be an entrypoint that has no modules of its own, but has an HTML import which uses modules...